### DISCUSSION TITLE
StaticTextBlock IsHtml=true : what are the style limits?

### DISCUSSION THREAD

**Discussion by LSNicholls** [serenity-is/Serenity#6634] [Help] [answered]
Hi all,

I am using HtmlContentEditor to create and store some richly-formatted text, and then attempting to display it using StaticTextBlock with IsHtml = true in a form.

I have noticed that the StaticTextBlock does not reproduce all the types of formatting that HtmlContentEditor can supply.  One item I know for sure isn't coming across is  blockquote .  It looks fine when being edited but in the form with StaticTextBlock, it's ignored.
I'll put an example below.

So, I am wondering:  

- [ ] is this a known limitation of StaticTextBlock, and I should be using one of the other editors, such as HtmlNoteContentEditor, if I intend to display with StaticTextBlock? or
 
- [x] should this work, but somehow the CSS accessible to the form is screwing it up?  this seems like a real possibility, considering that the blockquote tag is there in the text displayed by the form, when I inspect the source. If this is the case, what can I do to make sure that the user receives all the benefit of the editor when the content is viewed? -- or
 
- [ ]  other ideas?

Thank you.  And, because I haven't done a systematic check of *all* kinds of formatting that the editor can supply, compared with what comes across in the form, please let me know if you would like me to do that.

Here's the example -- first, what it looks like in the editor -- the editor tag is completely simple, just  [HtmlContentEditor(Rows = 8)]:

![image](https://user-images.githubusercontent.com/88156108/210196553-38092876-529f-4a6e-9890-4b83e6bc9526.png)

second, here is what is stored in the database based on this edit:

```
<p>More content here with <strong>normal formatting.</strong></p>
<p>Here's a blockquote:</p>
<blockquote>
<p>The GQC application is built using the <a href="https://serenity.is/">Serenity</a> framework.</p>
</blockquote>
```

now, what it looks like in the resulting form.  There is nothing at all complicated about the form, besides the IsHtml=true attribute the StaticTextBlock only has HideLabel=true, and the form has nothing else at all.

![image](https://user-images.githubusercontent.com/88156108/210196582-9a7576a3-2c52-487d-84eb-b2c717761a81.png)

**Comment by LSNicholls** [comment]
Update:  I figured out that it is indeed the CSS and the only way I can see to do it, so far,  is 

```
/* ltr only version */
.s-XXXXDialog blockquote {
     font-style: italic;
      /* font-family: Georgia, Times, "Times New Roman", serif; */
     padding: 2px 0;
     border-style: solid;
     border-color: #ccc;
     border-width: 0;
     padding-left: 20px;
     padding-right: 8px;
     border-left-width: 5px;
}

```
... this is not ideal for several reasons, not counting the LTR-only-so-far resolution:

1.  It doesn't guarantee WYSIWYG for anything but blockquote so I will indeed have to experiment to find out everything that might not work as the user might expect, based on the editor's capabilities.  
2. I've hard-coded the values because I don't know enough about referencing another class somewhere in CSS to do it otherwise.  I would love to know how to do this, even though I'm not sure it would be suitable, because I wouldn't necessarily want (say) the font-family in the original to come through (I've taken out that line, above).  
3. I would have liked to have done this using use FormCssClass or CssClass attributes on the StaticTextBlock, but couldn't figure out how to do that in a way that would allow me to reference different possible types of child elements.  This might make it easier to handle the problem in multiple forms, although I really only have one.

... So... I have a resolution, but any thoughts on what the "correct" way to do this would be???

Thanks to anybody for advice.

**Comment by wezmag** [comment, marked as answer]
The content in the `HtmlContentEditor` uses only styles from `wwwroot\Content\site\site.htmlcontent.css`. ([source code here](https://github.com/serenity-is/Serenity/blob/master/src/Serenity.Scripts/corelib/src/ui/editors/htmlcontenteditor.ts#L124))
The content in the `StaticTextBlock` uses all styles from CSS files included in the `appsettings.bundles.json`.
That's why they are not the same.

To make them look the same, you need to make sure both content in the `HtmlContentEditor` and `StaticTextBlock` are using the same set of CSS files.

You can find more information in the CKEditor website
1. https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-contentsCss
2. https://ckeditor.com/docs/ckeditor4/latest/features/styles.html

**Reply by LSNicholls** [reply]
Thank you Wesley.  I didn't think of fixing it from the CKEditor side.
